sfge/src/main/java/com/salesforce/rules/AbstractPathBasedRule.java

```diff
@@ -15,4 +18,12 @@ public abstract class AbstractPathBasedRule extends AbstractRule {
     public Optional<PathExpansionObserver> getPathExpansionObserver() {
         return Optional.empty();
     }
+
+    /** Allows rules to indicate what types of path sources they are interested in. */
+    public abstract List<ApexPathSource.Type> getSourceTypes();
```

Comment:
Path-based rules can implement this method to specify which specific sources they're interested in.

Reply:
Can this be a set so that the elements are always unique?